aep/general/0124/aep.md.j2

```diff
@@ -78,8 +77,54 @@ recommended if additional metadata is required in the relationship, or if the
 restrictions around the use of a repeated field preclude the use of that
 approach.
 
+### Embedded resources
+
+Resource references as described use string references rather than embedding
+one resource inside another because embedding resources can lead to a number of
+issues. Retrieving resources with arbitrarily deep nesting may require multiple
+serial internal RPCs, leading to complex service dependencies latency and
+reliability issues.
+
+However, sometimes dereferencing resource references is useful, or even
+necessary. For example, an API may wish to allow filtering of a resource based
+on a field of a resource it references. In other cases, it may be much cheaper
+and/or faster for the server to perform the dereferencing than for the client
+to make multiple serial requests.
+
+In these cases, an alternative resource association pattern **may** be used,
+where the resource reference field is not a string, but is instead the
+referenced resource itself.
+
+The default behavior **should** be to populate only the `path` field of the
```

Comment:
wouldn't this erode the useful of the embedded resource quite a bit? for example, I could imagine for SDKs where a required property is expressed as a required value in the initializer of a struct / class, you'll have errors when constructing that object.

I think it's not ideal to use a separate class here, because if you do that it'll make it difficult to pass that object to other SDK calls.

Reply:
That's an interesting angle that I hadn't considered, and I don't immediately have a great answer to it.

However, I don't think it's really about only the `path` field being populated on responses.  The guidance should probably also say something about how requests should not set any other fields than `path` with an embedded resource reference.

The intention behind this PR is purely to make reading/dereferencing resource references more convenient.  It's definitely not intended that e.g. `UpdateBook` would let you update `Book.author.name`, and I'm kind of inclined to write guidance that says APIs **must not** allow this.  (But maybe it should be a `**should not**` - perhaps some API really wants to do it, and it's not actually harmful?  It does seem like somewhat of a slippery slope towards violating the overall resource-oriented design philosophy.)

Reply:
Regarding generated classes in client libraries:  I actually think maybe it's wrong for a client library to require fields with the `REQUIRED` field behavior to be set at object construction time.  It seems analogous to the classic go/required-fields-considered-harmful - no?

Reply:
We discussed this offline and concluded that the `PATCH` case requires more flexible objects, and so can call this point resolved.